### PR TITLE
add template replace option

### PR DESCRIPTION
**Pull request by pitwegner**
Signed-off-by: pitwegner <pit.wegner@gmail.com>

## Hey, I just made a Pull Request!

Allow the `fetch:template` action to overwrite existing files. This is useful for adding changes to an existing git repo.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by pitwegner** [comment]
Not sure how to test this with the current testing framework. seems like there is no way to specify already existing files in the working directory.

**Comment by benjdlambert** [comment]
Hey :wave: thanks for this!

For testing we use `mock-fs`, and I think what you will want to do is add a record to this object which has the key of the path in the workspace and the contents as the value. 

https://github.com/backstage/backstage/blob/fe3e4bfcf4852856fea76af526b08d19143158cf/plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/template.test.ts#L98-L100

And then assert that when replace is false it's the same value and when replace is true it replaced the contents of it.

Some more docs on [mock-fs](https://github.com/tschaub/mock-fs)

File: .changeset/perfect-cheetahs-serve.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-scaffolder-backend': patch
```
Comment (benjdlambert) [inline .changeset/perfect-cheetahs-serve.md:2]: Should be a minor change as this package is `>1.0.0` :pray:

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Just one thing about the changeset and I think we're good to go! :pray:

**Review by benjdlambert — APPROVED** [review]

**Comment by benjdlambert** [comment]
Nice! Thanks for this!

**Pull request merged**